teps/0056-pipelines-in-pipelines.md

```diff
@@ -0,0 +1,368 @@
+know that this has succeeded?
+-->
+
+- Defining a set of `Tasks` as a complete unit of execution
+- Decoupling failures in unrelated sets of `Tasks`
+
+### Non-Goals
+
+<!--
+What is out of scope for this TEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Ignoring `Task` failures, which is addressed in [TEP-0050 Ignore Task Failures](https://github.com/tektoncd/community/blob/main/teps/0050-ignore-task-failures.md)
+- Composing `Tasks` in `Tasks`, which is addressed in [TEP-0044 Composing Tasks with Tasks](https://github.com/tektoncd/community/pull/316)
+
+### Use Cases
+
+<!--
+Describe the concrete improvement specific groups of users will see if the
+Motivations in this doc result in a fix or feature.
+
+Consider both the user's role (are they a Task author? Catalog Task user?
+Cluster Admin? etc...) and experience (what workflows or actions are enhanced
+if this problem is solved?).
+-->
+
+* Reusability and Composability
+  
+As a `Pipeline` author, I need to define a set of `Tasks` as a complete 
+unit of execution that I can share across `Pipelines`.
+  
+For example, I can define and distribute `linting` set of `Tasks` and `testing` set of `Tasks`:
+```
+        linting                     
+           |                           
+           v
+        testing
+```
+
+where `linting` is made up of `lint` and `report-linter-output` tasks:
+``` 
+          lint                     
+           |                           
+           v
+   report-linter-output 
+```
+
+and `testing` is made up of `unit-tests`, `integration-tests` and `report-test-results` `Tasks`:
+```                           
+       unit-tests
+           |                           
+           v                           
+   integration-tests 
+           |
+           v 
+  report-test-results
+```
+
+* Failure Strategies
+  
+As a `Pipeline` author, I need to decouple failures in unrelated sets of `Tasks`.
+
+For example, I can design a `Pipeline` where `lint`might fail, but the `unit-tests`, `integration-tests` and
+`report-test-results` will continue executing so that I can get the test results without a rerun of the `Pipeline`:
+
+```
+          lint                     unit-tests
+           |                           |
+           v                           v
+   report-linter-output        integration-tests 
+                                       |
+                                       v 
+                              report-test-results
+```
+
+## Requirements
+
+<!--
+Describe constraints on the solution that must be met. Examples might include
+performance characteristics that must be met, specific edge cases that must
+be handled, or user scenarios that will be affected and must be accomodated.
+-->
+
+- Users should be able to define and distribute a set of `Tasks` as a complete unit of execution
+- Users should be able decouple failures in unrelated sets of `Tasks`
```

Comment:
Do you think we should add a requirement here that the `results` and `condition` of the sub-pipeline should be made available in the parent pipeline?  For example I could run two sub-pipelines and then use a finally task to report results from both of them if they ran to success.

Reply:
Another possible requirement (although this might be verging into implementation detail and too much info right now?): Deleting a Pipeline with PiPs also deletes the PipelineRuns that were created for those PiPs.  This would mean users don't have to go back and clean up the PipelineRuns that were created for PiPs - they'd all get deleted when the root `Run` or `Pipeline` is deleted.

Reply:
added those and a couple more requirements about the main pipeline and the subpipeline 

thanks @sbwsg 🙏🏾